e2e/pageobjects/ide/Editor.ts

```diff
@@ -179,10 +230,126 @@ export class Editor {
+
+        await this.driverHelper.waitVisibility(debugBreakpointLocator, timeout);
+    }
+
+    async waitBreakpointAbsence(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointLocator: By = await this.getDebugBreakpointLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(debugBreakpointLocator, timeout);
+    }
+
+    async waitBreakpointHint(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointHintLocator: By = await this.getDebugBreakpointHintLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitVisibility(debugBreakpointHintLocator, timeout);
+    }
+
+    async waitBreakpointHintDisappearance(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointHintLocator: By = await this.getDebugBreakpointHintLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(debugBreakpointHintLocator, timeout);
+    }
+
+    async activateBreakpoint(tabTitle: string,
+        lineNumber: number,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+
+        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+
+        for (let i = 0; i < attempts; i++) {
+            const elementLocator: By = await this.getLineNumberBlockLocator(tabTitle, lineNumber);
+            const element: WebElement = await this.driverHelper.waitVisibility(elementLocator, timeout);
+
+            try {
+                await this.driverHelper.getAction().mouseMove(element, { x: 5, y: 5 }).perform();
+                await this.waitBreakpointHint(tabTitle, lineNumber);
+                await this.driverHelper.getAction().click().perform();
+                await this.waitBreakpoint(tabTitle, lineNumber);
+                return;
+            } catch (err) {
+                // ignore errors and wait
+                await this.driverHelper.wait(polling);
+            }
+        }
+
+        throw new Error(`Exceeded maximum breakpoint activation attempts`);
+    }
+
+    private async getLineYCoordinates(lineNumber: number): Promise<number> {
+        const lineNumberLocator: By = By.xpath(`//div[contains(@class, 'line-numbers') and text()='${lineNumber}']` +
+            `//parent::div[contains(@style, 'position')]`);
+
+        let elementStyleValue: string = await this.driverHelper.waitAndGetElementAttribute(lineNumberLocator, 'style');
+
+        elementStyleValue = elementStyleValue.replace('position: absolute; top: ', '');
+        elementStyleValue = elementStyleValue.replace('px; width: 100%; height: 19px;', '');
+
+        const lineYCoordinate: number = Number.parseInt(elementStyleValue, 10);
```

Comment:
Maybe using regex would be nicer here? I'm not sure how regexes work in typescript... I've just tried som online javascript like https://playcode.io/online-javascript-editor and this seems to work ok:
```
var str = 'position:absolute;top:627px;width:100%;height:19px;';
console.log(/(?<=top:)(\d*)/.exec(str)[0]); \\prints 627
```